### PR TITLE
[DOCS] Test examples in Graph API

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/issues/30665

This PR enables testing of the examples in the Graph API reference.

### CODE REVIEW

File: x-pack/docs/build.gradle [hunk 1]
```diff
@@ -737,3 +736,19 @@ setups['sensor_prefab_data'] = '''
...
+            number_of_shards: 1
+            number_of_replicas: 0
+          mappings:
+            _doc:
+              properties:
+                product:
+                  type: keyword
+                query_time:
+                  type: date
+'''
```
Comment (markharwood) [inline x-pack/docs/build.gradle]: Needs some data adding in here?

File: x-pack/docs/en/rest-api/graph/explore.asciidoc [hunk 2]
```diff
@@ -204,6 +208,7 @@ POST clicklogs/_xpack/graph/_explore
 }
 --------------------------------------------------
 // CONSOLE
+// TEST[setup:clicklogs_index]
 <1> Seed the exploration with a query. This example is searching
 clicklogs for people who searched for the term "midi".
 <2> Identify the vertices to include in the graph. This example is looking for
```
Comment (markharwood) [inline x-pack/docs/en/rest-api/graph/explore.asciidoc]: I see this query refers to a "query.raw" which is a field not likely to be in the clicklogs index mapping as it is currently defined in the build.gradle file

**Comment by markharwood** [comment]
I saw a couple of things in the setup - missing data and fields

File: x-pack/docs/en/rest-api/graph/explore.asciidoc [hunk 3]
```diff
@@ -56,7 +56,7 @@ Elasticsearch query. For example:
   }
 }
 --------------------------------------------------
-
+// NOTCONSOLE
```
Comment (nik9000) [inline x-pack/docs/en/rest-api/graph/explore.asciidoc]: I think we'd be much better off writing a real request with just the `query` part in it or something like that. We had a lot of "unrooted" JSON in our docs before we had the docs tests and removing it all made the docs a ton more clear. No more wondering "where does this go"? Also, the way this is it doesn't get any test coverage.

Comment (markharwood) [reply x-pack/docs/en/rest-api/graph/explore.asciidoc]: I recall the response parts were tricky to test given the scores produced and how they might change with Lucene versions etc. and the difficulty involved in masking them from any "actual vs expected" result comparisons.

Comment (nik9000) [reply x-pack/docs/en/rest-api/graph/explore.asciidoc]: It is possible to ignore the numbers with tricks like `// TESTRESPONSE[s/12.3/$body.$_path/]`.

**Comment by lcawl** [comment]
Hi Mark,

Do you have any suggestions for what the index mapping should be to make this work?  I tried using https://github.com/elastic/x-pack-elasticsearch/blob/master/plugin/src/test/resources/rest-api-spec/test/graph/10_basic.yml <https://github.com/elastic/x-pack-elasticsearch/blob/master/plugin/src/test/resources/rest-api-spec/test/graph/10_basic.yml> steps, but it also returns empty results.  If it’s not possible to create a simple example with meaningful results for test purposes, should we disable code snippet testing for this page?

Best regards,

Lisa Cawley | Senior Technical Writer | Elastic
lcawley@elastic.co

**Comment by colings86** [comment]
@markharwood @lcawl Is this still something we want to do? If so what do we need to do to progress it?

**Comment by markharwood** [comment]
>If so what do we need to do to progress it?

I can take a closer look at Nik's suggestion for masking scores but the elements returned are also liable to change order too. I'm not sure the path expressions for string replacements would be expressive enough to cope with this.
The challenge here was always the reliance on the doc framework's strict string-based comparisons of expected vs actual on fully serialized results. Robust tests are hard to configure when trying to patch the variations in result strings

**Pull request closed without merging**